util/topgen/validate.py

```diff
@@ -95,6 +95,11 @@
     'freq': ['s', 'frequency of clock in Hz'],
 }
 
+clock_srcs_optional = {
+    'derived': ['s', 'whether clock is derived'],
+    'params': ['s', 'extra clock paramters']
```

Comment:
nit: paramters -> parameters

Reply:
i am the king of typos... thank you for catching these.